pointgrey_camera_driver/src/PointGreyCamera.cpp

```diff
@@ -47,1107 +52,845 @@ PointGreyCamera::~PointGreyCamera()
+  try
   {
-    throw CameraTimeoutException("PointGreyCamera: Failed to retrieve buffer within timeout.");
+    // Activate chunk mode
+    //
+    // *** NOTES ***
+    // Once enabled, chunk data will be available at the end of the payload
+    // of every image captured until it is disabled. Chunk data can also be
+    // retrieved from the nodemap.
+    //
+    Spinnaker::GenApi::CBooleanPtr ptrChunkModeActive = nodeMap.GetNode("ChunkModeActive");
+    if (!Spinnaker::GenApi::IsAvailable(ptrChunkModeActive) || !Spinnaker::GenApi::IsWritable(ptrChunkModeActive))
+    {
+      ROS_ERROR_STREAM("Unable to activate chunk mode. Aborting...");
+      return -1;
+    }
+    ptrChunkModeActive->SetValue(true);
+    ROS_INFO_STREAM("Chunk mode activated...");
+
+    // Enable all types of chunk data
+    //
+    // *** NOTES ***
+    // Enabling chunk data requires working with nodes: "ChunkSelector"
+    // is an enumeration selector node and "ChunkEnable" is a boolean. It
+    // requires retrieving the selector node (which is of enumeration node
+    // type), selecting the entry of the chunk data to be enabled, retrieving
+    // the corresponding boolean, and setting it to true.
+    //
+    // In this example, all chunk data is enabled, so these steps are
+    // performed in a loop. Once this is complete, chunk mode still needs to
+    // be activated.
+    //
+    Spinnaker::GenApi::NodeList_t entries;
+    // Retrieve the selector node
+    Spinnaker::GenApi::CEnumerationPtr ptrChunkSelector = nodeMap.GetNode("ChunkSelector");
+    if (!Spinnaker::GenApi::IsAvailable(ptrChunkSelector) || !Spinnaker::GenApi::IsReadable(ptrChunkSelector))
+    {
+      ROS_ERROR_STREAM("Unable to retrieve chunk selector. Aborting...");
+      return -1;
+    }
+    // Retrieve entries
+    ptrChunkSelector->GetEntries(entries);
+
+    ROS_INFO_STREAM("Enabling entries...");
+
+    for (int i = 0; i < entries.size(); i++)
+    {
+      // Select entry to be enabled
+      Spinnaker::GenApi::CEnumEntryPtr ptrChunkSelectorEntry = entries.at(i);
+      // Go to next node if problem occurs
+      if (!Spinnaker::GenApi::IsAvailable(ptrChunkSelectorEntry) || !Spinnaker::GenApi::IsReadable(ptrChunkSelectorEntry))
+      {
+        continue;
+      }
+      ptrChunkSelector->SetIntValue(ptrChunkSelectorEntry->GetValue());
+
+      ROS_INFO_STREAM("\t" << ptrChunkSelectorEntry->GetSymbolic() << ": ");
+      // Retrieve corresponding boolean
+      Spinnaker::GenApi::CBooleanPtr ptrChunkEnable = nodeMap.GetNode("ChunkEnable");
+      // Enable the boolean, thus enabling the corresponding chunk data
+      if (!Spinnaker::GenApi::IsAvailable(ptrChunkEnable))
+      {
+        ROS_INFO("Node not available");
+      }
+      else if (ptrChunkEnable->GetValue())
+      {
+        ROS_INFO("Enabled");
+      }
+      else if (Spinnaker::GenApi::IsWritable(ptrChunkEnable))
+      {
+        ptrChunkEnable->SetValue(true);
+        ROS_INFO("Enabled");
```

Comment:
I'd change a little the message, so you can distinguish it from the previous one. Not sure if important.